packages/@biomejs/tailwindcss-config-analyzer/src/generate-tailwind-preset.ts

```diff
@@ -0,0 +1,87 @@
+import { introspectTailwindConfig } from "./introspect.js";
+import { SortConfig, sortConfigFromSpec } from "./sort-config.js";
+import { readPackageUp } from "read-package-up";
+import path from "node:path";
+
+const ROOT_PACKAGE_NAME = "@biomejs/monorepo";
+const OUTPUT_PATH =
+	"crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/tailwind_preset.rs";
+
+const EXCLUDED_LAYERS = ["defaults", "base"];
+const LAYER_ORDER = ["components", "utilities"];
+
+const LINE_LIMIT = 100;
+const INDENT = "    ";
+const FILE_HEADER = `//! DO NOT EDIT MANUALLY - this file is autogenerated from the default Tailwind CSS config.
+//! To update, run \`pnpm gen:tailwind-preset\` from \`packages/@biomejs/tailwindcss-config-analyzer\`.
+
+use super::sort_config::UtilityLayer;
+`;
+
+function generateLayer({ layer, classes }: SortConfig["utilities"][number]) {
+	const header = `const ${layer.toUpperCase()}_LAYER_CLASSES: [&str; ${classes.length}] = [`;
+
+	// try single line
+	const singleLine = `${header}${classes.map((c) => `"${c}"`).join(", ")}];`;
+	if (singleLine.length < LINE_LIMIT) return `${singleLine}\n`;
+
+	// if single line is too long, do multi-line
+	return `${header}\n${classes.map((c) => `${INDENT}"${c}"`).join(",\n")},\n];\n`;
+}
+
+function generateLayerArray(layers: SortConfig["utilities"]) {
+	let output = `pub const TAILWIND_LAYERS: [UtilityLayer; ${layers.length}] = [\n`;
+	for (const { layer } of layers) {
+		output += `${INDENT}UtilityLayer {\n`;
+		output += `${INDENT}${INDENT}name: "${layer}",\n`;
+		output += `${INDENT}${INDENT}classes: ${layer.toUpperCase()}_LAYER_CLASSES.as_slice(),\n`;
+		output += `${INDENT}},\n`;
+	}
+	output += "];\n";
+	return output;
+}
+
+function generateFile(sortConfig: SortConfig) {
+	let output = FILE_HEADER;
+	output += "\n";
+	output += sortConfig.utilities.map(generateLayer).join("\n");
+	output += "\n";
+	output += generateLayerArray(sortConfig.utilities);
+	return output;
+}
+
+async function findRoot() {
+	let nextPath = import.meta.dir;
```

Comment:
This is extremely specific to Bun. It won't work with Node.js

Reply:
This is internal codegen code and is not intended to work with Node.js. Bun is used as an alternative to something like tsx or ts-node here, so we can use their handy APIs. Another reason why I chose Bun for this task is because they support "auto-install" which could be useful in the future to run the script with the latest tailwindcss package without the need to handle the dependency manually.

Reply:
We can achieve the same outcome without using Bun specific API. It will make switching easier if needed. Can you please change this?

Reply:
Don't see a reason why we'd want to switch but sure, I'll do it the Node.js way.

Reply:
Since I'm already using other Bun APIs (like `Bun.write`), I'm gonna leave it like this. Should be easy enough to switch if a reason for it comes up in the future.